googletest/src/gtest.cc

```diff
@@ -800,41 +807,50 @@ std::string UnitTestImpl::CurrentOsStackTraceExceptTop(int skip_count) {
       );  // NOLINT
 }
 
+// returns the monotonic time in milliseconds
+TimeInMillis GetMonotonicTimeInMillis() {
+#if GTEST_LANG_CXX11
+  const std::chrono::steady_clock::duration now{ std::chrono::steady_clock::now().time_since_epoch() };
+  const std::chrono::milliseconds nowMs{ std::chrono::duration_cast<std::chrono::milliseconds>(now) };
+  return static_cast<TimeInMillis>(nowMs.count());
+#elif GTEST_OS_WINDOWS 
+  static const long long _Freq = _Query_perf_frequency(); // doesn't change after system boot
+  const long long _Ctr = _Query_perf_counter();
+  const long long _Whole = (_Ctr / _Freq);
+  const long long _Part = (_Ctr % _Freq);
+  return static_cast<TimeInMillis>(_Whole * 1000 + (_Part * 1000) / _Freq);
+#elif GTEST_OS_LINUX
+  timespec now;
+  clock_gettime(CLOCK_MONOTONIC, &now);
+  return static_cast<TimeInMillis>(now.tv_sec) * 1000 + static_cast<TimeInMillis>(now.tv_nsec / 1000000);
+#elif GTEST_OS_MAC
+  static bool initTimeBase = true;
+  static mach_timebase_info_data_t timebaseInfo;
```

Comment:
I'm not very fluent with Mac API, but can't you use timebaseInfo.numer/denom to check whethet it is initialized ?  and spare static bool ?

Reply:
I believe that would be valid.  Will change.